src/Compilers/Core/Portable/NativePdbWriter/ComMemoryStream.cs

```diff
@@ -52,6 +53,40 @@ public void CopyTo(Stream stream)
             }
         }
 
+        public IEnumerable<ArraySegment<byte>> GetChunks()
+        {
+            byte[] lazyZeroChunk = null;
+
+            int chunkIndex = 0;
+            int remainingBytes = _length;
+            while (remainingBytes > 0)
+            {
+                int bytesToCopy = Math.Min(ChunkSize, remainingBytes);
+
+                byte[] chunk;
+                if (chunkIndex < _chunks.Count)
+                {
+                    chunk = _chunks[chunkIndex++];
+                }
+                else if (bytesToCopy == ChunkSize)
```

Comment:
When would this `if` block get hit?

Reply:
If the writer to the stream seeks beyond the currently allocated chunks and then doesn't write to that "reserved" space. 



---
In reply to: [136211645](https://github.com/dotnet/roslyn/pull/21712#discussion_r136211645) [](ancestors = 136211645)